internal/raft/append_entries_test.go

```diff
@@ -0,0 +1,77 @@
+package raft
+
+import (
+	"net"
+	"testing"
+
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+	"github.com/tomarrell/lbadd/internal/network"
+	"github.com/tomarrell/lbadd/internal/raft/cluster"
+	"github.com/tomarrell/lbadd/internal/raft/message"
+)
+
+func TestAppendEntries(t *testing.T) {
+	assert := assert.New(t)
+
+	log := zerolog.Nop()
+	cluster := cluster.NewTCPCluster(log)
+
+	conn1, conn2 := net.Pipe()
+	conn3, conn4 := net.Pipe()
+	tcp1int, tcp1ext := network.NewTCPConn(conn1), network.NewTCPConn(conn2)
+	tcp2int, tcp2ext := network.NewTCPConn(conn3), network.NewTCPConn(conn4)
+	defer func() {
+		_ = tcp1int.Close()
+		_ = tcp1ext.Close()
+		_ = tcp2int.Close()
+		_ = tcp2ext.Close()
+	}()
+	cluster.AddConnection(tcp1int)
+	cluster.AddConnection(tcp2int)
+
+	node := &Node{
+		State: StateFollower.String(),
+		PersistentState: &PersistentState{
+			CurrentTerm: 0,
+			VotedFor:    nil,
+			SelfID:      cluster.OwnID(),
+			PeerIPs:     cluster.Nodes(),
+		},
+		VolatileState: &VolatileState{
+			CommitIndex: -1,
```

Comment:
Please document better, what -1 exactly stands for.
Is it a valid value? Do we try to break something with this test?

Reply:
Why was this resolved? It isn't resolved

Reply:
comment added